clang-tools-extra/include-cleaner/lib/WalkAST.cpp

```diff
@@ -146,9 +148,24 @@ class ASTWalker : public RecursiveASTVisitor<ASTWalker> {
     //
     // If it's an enum constant, it must be due to prior decl. Report references
     // to it when qualifier isn't a type.
-    if (llvm::isa<EnumConstantDecl>(FD)) {
-      if (!DRE->getQualifier() || DRE->getQualifier()->getAsNamespace())
-        report(DRE->getLocation(), FD);
+    auto QualifierIsNamepsaceOrNone = [&DRE]() {
```

Comment:
NIT: Maybe make this a helper function instead?
Having a lambda reduces the scope of visibility, but we also loose the benefits of making the body of our function smaller. I tend to prefer smaller functions, but up to you.